### PR TITLE
Added sleep mode

### PR DESCRIPTION
**Pull request by wsyxbcl**
use sleeptime and waketime for sleep mode.
// first time using javascript, please point out if there's something stupid...

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -144,6 +157,38 @@ bot.onText(/^\/autodelete(@sticker_time_bot)?(\s+([^\s]+))?$/, (msg, match) => {
     }
 });
 
+bot.onText(/\/sleeptime (\d+)/, (msg, match) => {
+    const chatId = msg.chat.id;
+    // bot.sendMessage(chatId, match[0]+'  '+match[1]+'  '+match[2]+'  '+match[3])
+    if (match[1]) {
+        if (match[1] <= 60 && match[1] >= 0){
```
Comment (imxieyi) [inline index.js]: Hour should never be larger than 23.
```suggestion
        if (match[1] <= 23 && match[1] >= 0){
```

File: index.js [hunk 2]
```diff
@@ -144,6 +157,38 @@ bot.onText(/^\/autodelete(@sticker_time_bot)?(\s+([^\s]+))?$/, (msg, match) => {
...
+            bot.sendMessage(chatId, 'Invalid time, 0-23 expected');
+            bot.sendSticker(chatId, stickers[13])
+        }
+    }
+});
+
+bot.onText(/\/waketime (\d+)/, (msg, match) => {
+    const chatId = msg.chat.id;
+    if (match[1]) {
+        if (match[1] <= 60 && match[1] >= 0){
```
Comment (imxieyi) [inline index.js]: Here again.
```suggestion
        if (match[1] <= 23 && match[1] >= 0){
```

File: index.js [hunk 3]
```diff
@@ -144,6 +157,38 @@ bot.onText(/^\/autodelete(@sticker_time_bot)?(\s+([^\s]+))?$/, (msg, match) => {
...
+            logger.info(chatId + ' set sleeptime to '+match[1]+':00');
+            bot.sendMessage(chatId, 'Set sleeptime to '+match[1]+':00');
+            data.sleeptime[chatId] = match[1]
+            saveData();
+        } else {
+            bot.sendMessage(chatId, 'Invalid time, 0-23 expected');
+            bot.sendSticker(chatId, stickers[13])
+        }
+    }
```
Comment (imxieyi) [inline index.js:168]: You should give some feedback to the user instead of just ignoring invalid inputs.

File: index.js [hunk 4]
```diff
@@ -32,7 +32,10 @@ const stickers = [
     'CAADBQAD-wADDxXNGe8aqxEu9OCLAg',
     'CAADBQAD_AADDxXNGSb44I6FN-UzAg',
     'CAADBQAD_QADDxXNGYGYV17DXBbkAg',
-    'CAADBQAD_gADDxXNGWuj_Z6psGN4Ag'
+    'CAADBQAD_gADDxXNGWuj_Z6psGN4Ag',
+    'CAADBQADCAAD7Dx2Ikbxv7doe-meAg', // still working!
+    'CAADBQADDAAD7Dx2IpKz4vfpPlrCAg', // stupid human!
+    'CAADBQADBQAD7Dx2IlqNSpG32Rg4Ag'  // get up!
```
Comment (imxieyi) [inline index.js]: I don't think it make much sense to send `still working` and `stupid human` stickers.

**Comment by imxieyi** [comment]
The sleep/wake time function makes sense to me. Why did you close this PR?

File: index.js (hunk 4 above)
Comment (wsyxbcl) [reply index.js]: Feel free to adjust it by your idea, I used another branch to request a pull.

File: index.js (hunk 1 above)
Comment (wsyxbcl) [reply index.js]: Yeah, used minute for demo and forget to change back

File: index.js (hunk 4 above)
Comment (wsyxbcl) [reply index.js]: This part is just for fun...That's why I decided to make another branch and keep this one more 'official'

**Comment by imxieyi** [comment]
After testing it works fine on my side. I will merge this.

**Comment by imxieyi** [comment]
This feature is now available in the public bot.

**Pull request merged**